### DISCUSSION TITLE
HW 07 Problem 3 Result Issue

### DISCUSSION THREAD

**Discussion by travismalave** [cooperrc/advanced-dynamics#47] [General]
Hi Everyone,
Has anyone gotten results for Problem 3 that visually match what the results from the text show but are flipped upside down? If so, any feedback on what to updated within the code if you were able to correct it?
Thanks for the help!

**Comment by RyanW-UConn** [comment]
I had that problem at one point, in my case it was that I had flipped a sign on the Ct equation when I took the d/dt of the constraint equation.

**Reply by RyanW-UConn** [reply]
There was actually one other thing I forgot - I changed my q0 for theta2 to be -pi/3 instead of pi/3 and that turned things around. I'm not sure if this is completely correct though, Problem 3 doesn't specify an initial condition I don't think. At least not in the way Problem 1 does. I think this behavior might have to do with the fact there can be different branches possible with a slider/crank, so changing the initial guess might result in the fsolve finding a different (valid) answer - but I don't know if that is actually what is going on here.

**Reply by travismalave** [reply]
Thanks for the feedback! This help get my plot fixed for the most part.

The initial condition of -pi/3 makes sense when you look at the text this problem is based on. Ref. Shabana ch 3 Fig. 3.8, theta3 which is our theta 2. The way the angle is referenced makes it seem that the initial angle is something less than 2*pi,  (which aligns with the first plot for crank angle vs connecting rod showing ~6.05 rad). So, -pi/3 is reasonable assuming we are measuring the angle in the short distance (i.e., the negative direction).

And then flipping the sign of CT [8] then makes the plot match would be like you mentioned. Just trying to understand where this sign comes from. From the text CT [8] is just d/dt(F(t)) but there could be a floating negative somewhere in the code that got missed which would be hard to track down. Might just assume this is the case, but it would be good to know if it is and how/where the sign was lost.
